File: backend/src/main/scala/domains/message/http/response/MessageHistoryResponse.scala
package domains.message.http.response

import domains.message.model.*

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class MessageHistoryResponse(
  conversation: MessageConversationSummary,
  messages: List[DirectMessage],
  hasMore: Boolean,
  facts: ConversationMessageFacts
)

object MessageHistoryResponse:
  given Encoder[MessageHistoryResponse] = deriveEncoder[MessageHistoryResponse]
  given Decoder[MessageHistoryResponse] = deriveDecoder[MessageHistoryResponse]

File: frontend/src/features/message/model/response/MessageHistoryResponse.ts
import type { ConversationMessageFacts } from '@/features/message/model/response/ConversationMessageFacts'
import type { DirectMessage } from '@/features/message/model/response/DirectMessage'
import type { MessageConversationSummary } from '@/features/message/model/response/MessageConversationSummary'

export type MessageHistoryResponse = {
  conversation: MessageConversationSummary
  messages: DirectMessage[]
  hasMore: boolean
  facts: ConversationMessageFacts
}
